src/OpenTracing/Util/AsyncLocalScope.cs

```diff
@@ -0,0 +1,43 @@
+namespace OpenTracing.Util
+{
+    /// <summary>
+    /// The <see cref="AsyncLocalScope"/> is a simple <see cref="IScope"/> implementation
+    /// that relies on C#'s AsyncLocal/CallContext storage primitive.
+    /// </summary>
+    /// <seealso cref="IScopeManager"/>
+    public class AsyncLocalScope : IScope
+    {
+        private readonly AsyncLocalScopeManager _scopeManager;
+        private readonly ISpan _wrappedSpan;
+        private readonly bool _finishOnDispose;
+        private readonly IScope _scopeToRestore;
+
+        public AsyncLocalScope(AsyncLocalScopeManager scopeManager, ISpan wrappedSpan, bool finishOnDispose)
```

Comment:
[nit] it looks like the only place `wrappedSpan` is supplied here it's not wrapped in anything?

Reply:
Oooh it's the span that is BEING wrapped, not a span that IS wrapped. Obviously your naming is fine now that I realize this, but given I was confused for a bit suggests possibility for better naming (subjective, feel free to leave if you disagree)

Reply:
Naming is from Java, except that I named it `wrappedSpan` instead of just `wrapped`.